presto-main/src/main/java/com/facebook/presto/cost/JoinStatsRule.java

```diff
@@ -241,15 +242,10 @@ PlanNodeStatsEstimate calculateJoinComplementStats(
             PlanNodeStatsEstimate leftStats,
             PlanNodeStatsEstimate rightStats)
     {
-        // TODO: add support for non-equality conditions (e.g: <=, !=, >)
-        if (filter.isPresent()) {
-            // non-equi filters are not supported
-            return UNKNOWN_STATS;
-        }
-
         if (criteria.isEmpty()) {
             if (rightStats.getOutputRowCount() > 0) {
                 // all left side rows are matched
+                // TODO: account non-equi conditions here somehow
```

Comment:
maybe using the UNKNOWN_FILTER_COEFFICIENT, same as if there's a mix of equi and non-equi conditions.